### PR TITLE
Implement Doctrine with custom DBAL types for Value-objects

### PR DESCRIPTION
**Pull request by dennisdegreef**
@fvdb Any idea why it can't locate the `PullRequest.orm.yml` file?

`No mapping file found named 'PullRequest.orm.yml' for class 'LonelyPullRequests\Domain\PullRequest'.`

### CODE REVIEW

File: app/config/config.yml [hunk 1]
```diff
@@ -61,7 +61,15 @@ doctrine:
...
         auto_generate_proxy_classes: "%kernel.debug%"
-        auto_mapping: true
+        entity_managers:
+            default:
+                mappings:
+                    custom_mapping:
+                        type: yml
+                        prefix: LonelyPullRequests\Domain\
+                        dir: "%kernel.root_dir%/../src/Domain/"
+                        is_bundle: false
```
Comment (fvdb) [inline app/config/config.yml]: Should be true.

File: app/config/config.yml [hunk 2]
```diff
@@ -61,7 +61,15 @@ doctrine:
...
     orm:
         auto_generate_proxy_classes: "%kernel.debug%"
-        auto_mapping: true
+        entity_managers:
+            default:
+                mappings:
+                    custom_mapping:
+                        type: yml
+                        prefix: LonelyPullRequests\Domain\
+                        dir: "%kernel.root_dir%/../src/Domain/"
```
Comment (fvdb) [inline app/config/config.yml]: Should remove this.

File: app/config/config.yml (hunk 1 above)
Comment (fvdb) [inline app/config/config.yml]: Also add: ```alias: "LonelyPullRequestsBundle"```

Comment (dennisdegreef) [inline app/config/config.yml]: But i'm mapping to the Domain part, which is not a bundle right?

**Comment by fvdb** [comment]
Also, you should create the actual mapping file in the Resources/config/doctrine directory of the bundle.

**Comment by dennisdegreef** [comment]
I have, it's an empty file now, the exception being that it still can't find it... But it's in these commits

File: app/config/config.yml (hunk 1 above)
Comment (fvdb) [inline app/config/config.yml]: Your mapping config is part of the bundle.

Comment (dennisdegreef) [inline app/config/config.yml]: I get that the mapping and repository itself are in the bundle though

**Comment by dennisdegreef** [comment]
Kind of works now, but with basic types... Should create `Doctrine\DBAL\Types\Type` classes for every value-object?

**Comment by dennisdegreef** [comment]
Fixed everything, except that Travis breaks on a database connection, will look into fixing that.

**Comment by dennisdegreef** [comment]
Build passes, let's move this to master, unless you have any comments, @fvdb ? :)

**Comment by dennisdegreef** [comment]
I'm just typing on...

**Comment by dennisdegreef** [comment]
Rebased with master

**Pull request merged**